wdn/templates_5.0/js-src/notice.babel.js

```diff
@@ -0,0 +1,387 @@
+define(['uuid-gen'], (uuidv4) => {
+	let widget = {};
+	let notices = document.querySelectorAll('[data-widget="notice"]');
+	let fixedBottomExists = document.querySelector('[id*="unl-widget-fixedBottom-"]') ? true : false; // flag for checking if a fixedBottom notice has been previously added to the page set flag to true
+
+	notices = [].slice.call(notices);
+
+// standard classes based on what kind of notice, keep it to DCF classes for now
+	const noticeClasses = {
+		notify: ['dcf-notice', 'unl-notice-notify'],
+		success: ['dcf-notice', 'unl-notice-success'],
+		alert: ['dcf-notice', 'unl-notice--alert'],
+		fatal: ['dcf-notice', 'unl-notice--fatal'],
+	};
+
+	const noticeLocationClasses = {
+		current: ['uno'],
+		nav: ['foo'],
+		fixedBottom: ['bar', 'dcf-fixed', 'dcf-notice-fixedBottom', 'dcf-pin-bottom', 'dcf-pin-right', 'dcf-pin-left'],
+		fixedBottomLeft: ['zoink'] // TODO add option that goes 50% width on desktop when fixedBottom
+	};
+
+// default animations depending on associated locations
+// fixedBottom associated with slideUp animation. Everything else uses slideInScroll
+	const noticeAnimationClasses = {
+		slideInScroll: ['baz'],
+		slideUp: ['car'],
+	};0
+
+	const closeButtonClasses = ['dcf-absolute', 'dcf-pin-top', 'dcf-pin-right', 'dcf-mt-3', 'dcf-mr-3', 'dcf-btn', 'dcf-btn-tertiary', 'js-notice-toggle'];
+
+
+	/**
+	 *
+	 * Functions
+	 */
+
+	/**
+	 * @purpose - move element to be the first child of main
+	 * @param el - the element to be moved
+	 *
+	 */
+	function prependMain(el) {
+		const main = document.querySelector('main');
+		const firstChild = main.firstElementChild;
+		main.insertBefore(el, firstChild);
+	}
+
+	/**
+	 * @purpose - permanently closes the notice element
+	 * @param notice - notice to be closed
+	 *
+	 */
+	function closeNotice(notice) {
+
+		function hideNotice(e) {
+			if (e.propertyName !== "max-height") return;
+			notice.classList.add('dcf-d-none');
+			notice.removeEventListener('transitionend', hideNotice);
+			document.querySelector('main').focus(); // sending focus back to main
+		}
+
+		notice.addEventListener('transitionend', hideNotice);
+		notice.classList.add('dcf-notice-fixedBottom--close');
+
+		localStorage.setItem(notice.id, 'closed');
+	}
+
+	/**
+	 * @purpose - collapse message when collapse button is selected
+	 * @param el - notice to be closed
+	 * @param closeButton - the close button associated with this notice
+	 * @param title - title of notice
+	 * @param message - message of notice
+	 *
+	 */
+	function collapseExpandMessage(el, closeButton, title, message) {
+		// Find out if notice is expanded
+		let expanded = closeButton.getAttribute('aria-expanded') === "true" ? true : false;
+
+		if (expanded) {
+			// if expanded, collapse message
+			closeButton.innerText = "Expand";
+			title.classList.add('dcf-notice_title--collapse');
+			message.classList.add('dcf-notice_message--collapse');
+			console.log(el.id, 'true');
```

Comment:
More debug code. There are a few others that are still here.

Reply:
I personally prefer the BEM formatting, but since we've agreed to go all kebab case, can these class names be updated for consistency?

Reply:
removed them.

Reply:
Will get them fixed @skoolbus39